server/controllers/feeds.ts

```diff
@@ -0,0 +1,172 @@
+import * as express from 'express'
+import * as Bluebird from 'bluebird'
+import { getFormattedObjects } from '../helpers/utils'
+import { CONFIG, STATIC_PATHS } from '../initializers'
+import {
+  asyncMiddleware,
+  feedsValidator,
+  feedsQueryCleaner,
+  feedsMissingParameters,
+  paginationValidator,
+  setDefaultPagination,
+  setDefaultSort,
+  usersGetValidator,
+  videosAddValidator,
+  videosGetValidator,
+  videosRemoveValidator,
+  videosSearchValidator,
+  videosSortValidator,
+  videosUpdateValidator
+} from '../middlewares'
+import { VideoModel } from '../models/video/video'
+import { UserModel } from '../models/account/user'
+import { AccountModel } from '../models/account/account'
+import * as url from 'url'
+import { logger } from '../helpers/logger'
+// @ts-ignore
+const Feed = require('feed')
+
+const feedsRouter = express.Router()
+
+// multipurpose endpoint
+feedsRouter.use('/feeds/videos.(xml|json(1)?|rss(2)?|atom(1)?)',
+  feedsValidator,
+  feedsMissingParameters,
+  asyncMiddleware(generateFeed)
+)
+
+// ---------------------------------------------------------------------------
+
+export {
+  feedsRouter
+}
+
+// ---------------------------------------------------------------------------
+
+async function generateFeed (req: express.Request, res: express.Response, next: express.NextFunction) {
+  let feed = initFeed()
+
+  // should we limit results to an account?
+  // (beware, only user accounts have videos)
+  let isAccountFiltering = false
+  let accountId: number
+  let accountName: string
+  if (req.params.accountId ||
+      req.params.accountName ||
+      req.query.accountId ||
+      req.query.accountName) {
+    isAccountFiltering = true
+  }
+
+  if (req.query.accountId) {
+    accountId = req.query.accountId
+  } else if (req.query.accountName) {
+    const userNameToId = await UserModel.loadByUsername(req.query.accountName)
+    // we need the user id and not the account id, since only user accounts have videos
+    accountId = userNameToId.toFormattedJSON().id
+  }
+
+  let resultList: {
+    data: VideoModel[];
+    total: number;
+  }
+  if (isAccountFiltering) {
+    resultList = await Bluebird.resolve(VideoModel.listUserVideosForApi(
+      accountId,
+      req.query.start,
+      req.query.count,
+      req.query.sort
+    ))
+  } else {
+    resultList = await Bluebird.resolve(VideoModel.listForApi(
+      req.query.start,
+      req.query.count,
+      req.query.sort,
+      req.query.filter
+    ))
+  }
+
+  // adding video items to the feed, one at a time
+  resultList.data.forEach(video => {
+    let formattedAccount = video.toFormattedJSON().account
```

Comment:
`const formattedAccount = video.Account.toFormattedJSON()`

Reply:
`server/controllers/feeds.ts(91,36): error TS2339: Property 'Account' does not exist on type 'VideoModel'` :slightly_frowning_face:

Reply:
Sorry: `const formattedAccount = video.VideoChannel.Account.toFormattedJSON()`